Counterexamples/EulerSumOfPowers.lean

```diff
@@ -0,0 +1,345 @@
+/-
+Copyright (c) 2025 Snir Broshi. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Snir Broshi
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset.Defs
+import Mathlib.Data.Finset.Card
+import Mathlib.Data.PNat.Notation
+import Mathlib.NumberTheory.FLT.Basic
+
+/-!
+# Euler's sum of powers conjecture
+
+Euler's sum of powers conjecture says that at least n nth powers of positive integers
+are required to sum to an nth power of a positive integer.
+
+This was an attempt to generalize Fermat's Last Theorem,
+which is the special case of summing 2 nth powers.
+
+https://en.wikipedia.org/wiki/Euler%27s_sum_of_powers_conjecture
+http://euler.free.fr/
+-/
+
+
+namespace Counterexample
+
+abbrev sumOfPowersConjectureAux {α β γ : Type*}
+    [Zero α] [Zero β] [Zero γ] [Add γ] [HPow α ℕ γ] [HPow β ℕ γ] :=
+  ∀ (n : ℕ) (a : List α) (b : β),
+  n ≠ 0 → a ≠ ∅ → 0 ∉ a → b ≠ 0 → (a.map (· ^ n)).sum = b ^ n → n ≤ a.length
+
+abbrev sumOfPowersConjecture := @sumOfPowersConjectureAux ℕ ℕ ℕ _ _ _ _ _ _
+
+abbrev sumOfPowersConjectureInt := @sumOfPowersConjectureAux ℤ ℤ ℤ _ _ _ _ _ _
```

Comment:
These are all `: Prop`s, and so:
* Should have the `: Prop` to make this clear
* Should start with an upper-case `S`.